compose/compose.go

```diff
@@ -0,0 +1,100 @@
+package compose
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/compose-spec/compose-go/cli"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/flags"
+	"github.com/docker/compose-cli/pkg/api"
+	"github.com/docker/compose-cli/pkg/compose"
+	libstack "github.com/portainer/docker-compose-wrapper"
+)
+
+type ComposeDeployer struct {
+	configPath string
+}
+
+func NewComposeDeployer(configPath string) (libstack.Deployer, error) {
+	return &ComposeDeployer{configPath}, nil
+}
+
+// Up creates and starts containers
+func (deployer *ComposeDeployer) Deploy(projectName, host string, filePaths []string, envFilePath string) error {
```

Comment:
Should support passing in a context in method signature as first argument.
Example use-cases:
- The caller should be able to cancel mid-flight deployment (e.g. API request is cancelled)
- Can pass in context with a timeout to timeout the deployment if it takes too long

```go
func (deployer *ComposeDeployer) Deploy(ctx context.Context, projectName, host string, filePaths []string, envFilePath string) error {
  ...
  err = service.Up(ctx, project, api.UpOptions{})
  ...
}
```

Reply:
:+1: